publictask/backend/plugin/src/main/kotlin/com/ritense/valtimo/backend/plugin/service/PublicTaskService.kt

```diff
@@ -77,8 +71,30 @@ class PublicTaskService(
         return ResponseEntity(formHtml, HttpStatus.OK)
     }
 
-    fun completeUserTaskWithPublicTaskSubmission(submission: Map<String, Any>): ResponseEntity<String> {
-        // TODO: build it ;)
+    fun completeUserTaskWithPublicTaskSubmission(
+        publicTaskId: String,
+        submission: JsonNode
+    ): ResponseEntity<String> {
+
+        val publicTaskEntity = publicTaskRepository.getReferenceById(UUID.fromString(publicTaskId))
+
+        val camundaTask = try {
+            processLinkActivityService.openTask(publicTaskEntity.userTaskId)
+        } catch (e: Exception) {
+            return taskNotAvailableResponse(e)
+        }
+
+        val formSubmissionResult = defaultFormSubmissionService.handleSubmission(
+            processLinkId = camundaTask.processLinkId,
+            formData = submission,
+            documentId = "3b7d5a8a-d0dd-4a8b-b31a-27837cebbe6e",
```

Comment:
IS dit altijd de zelfde?

Reply:
oops... vergeten toe te voegen. scherp. ik moet nog de documentId ergens vandaan toveren